### PR TITLE
Support CPU-only tests #4 and without cuDNN tests.

### PR DESCRIPTION
**Pull request by okuta**
This PR supports CPU-only tests #4 and without cuDNN tests.

It uses Attribute selector plugin in nose.
http://nose.readthedocs.org/en/latest/plugins/attrib.html

nosetests -a '!gpu' : Run CPU-only tests.
nosetests -a '!cudnn' : Run without cuDNN tests.

### CODE REVIEW

File: tests/attr.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    def wrap(x):
+        for k in args:
+            setattr(x, k, True)
+        for k, v in kwargs.items():
+            setattr(x, k, v)
+        return x
+    return wrap
+
+gpu   = attr('gpu')
+cudnn = attr('gpu', 'cudnn')
```
Comment (beam2d) [inline tests/attr.py]: Move these decorators to somewhere in chainer package (e.g. by adding chainer.testing module) and avoid relative imports from the test codes.

File: tests/attr.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+def attr(*args, **kwargs):
+    """Decorator that adds attributes.
+
+    """
```
Comment (beam2d) [inline tests/attr.py]: One line docstring should be in one line: """Decorator that adds attributes."""

File: tests/functions_tests/test_basic_math.py [hunk 3]
```diff
@@ -13,6 +15,7 @@ def setUp(self):
         self.x2 = numpy.random.uniform(.5, 1, (3, 2)).astype(numpy.float32)
         self.gy = numpy.random.uniform(-1, 1, (3, 2)).astype(numpy.float32)
 
+    @attr.gpu
```
Comment (beam2d) [inline tests/functions_tests/test_basic_math.py]: Remove it since this is not a test function.

File: tests/functions_tests/test_embed_id.py [hunk 4]
```diff
@@ -16,6 +19,7 @@ def setUp(self):
         self.x  = numpy.array([0, 1, 0], dtype=numpy.int32)
         self.gy = numpy.random.uniform(-1, 1, (3, 2)).astype(numpy.float32)
 
+    @attr.gpu
```
Comment (beam2d) [inline tests/functions_tests/test_embed_id.py]: Remove it. This is not a testing function.

File: tests/functions_tests/test_parameter.py [hunk 5]
```diff
@@ -15,6 +17,7 @@ def setUp(self):
     def tearDown(self):
         del self.func
 
+    @attr.gpu
```
Comment (beam2d) [inline tests/functions_tests/test_parameter.py]: Remove it. This is not a testing function.

**Comment by beam2d** [comment]
@okuta I added some comments. After you fix them, I will merge.

**Comment by okuta** [comment]
Thank you for your comments. I fixed them.

**Comment by beam2d** [comment]
[LGTM](http://lgtm.in/i/o426uDWXp)

**Pull request merged**